### ISSUE TITLE
`no-self-use` (`PLR6301`) false positive on methods that call `super`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#6961]
```py
from typing_extensions import override


class A:
    def foo(self):
        ...


class B(A):
    @override
    def foo(self):
        ...

    def bar(self):  # error: PLR6301
        super().foo()
```

in this case the method cannot be converted to a static method, otherwise it would break the `super` call

**Comment by LaBatata101** [comment]
I can work on this one

**Comment by charliermarsh** [comment]
Thanks!
